opamp/observiq/observiq_packages_state_provider_test.go

```diff
@@ -0,0 +1,550 @@
+
+				assert.ErrorContains(t, err, "method not implemented")
+			},
+		},
+	}
+
+	for _, tc := range testCases {
+		t.Run(tc.desc, tc.testFunc)
+	}
+}
+
+func TestFileContentHash(t *testing.T) {
+	testCases := []struct {
+		desc     string
+		testFunc func(*testing.T)
+	}{
+		{
+			desc: "Not Implemented",
+			testFunc: func(*testing.T) {
+				logger := zap.NewNop()
+				p := &PackagesStateProvider{
+					logger: logger,
+				}
+
+				actual, err := p.FileContentHash("name")
+
+				assert.Nil(t, actual)
+				assert.ErrorContains(t, err, "method not implemented")
+			},
+		},
+	}
+
+	for _, tc := range testCases {
+		t.Run(tc.desc, tc.testFunc)
+	}
+}
+
+func TestUpdateContent(t *testing.T) {
+	testCases := []struct {
+		desc     string
+		testFunc func(*testing.T)
+	}{
+		{
+			desc: "Not Implemented",
+			testFunc: func(*testing.T) {
+				logger := zap.NewNop()
+				p := &PackagesStateProvider{
+					logger: logger,
+				}
+				var r io.Reader
+
+				err := p.UpdateContent(context.TODO(), "name", r, []byte("hash"))
+
+				assert.ErrorContains(t, err, "method not implemented")
+			},
+		},
+	}
+
+	for _, tc := range testCases {
+		t.Run(tc.desc, tc.testFunc)
+	}
+}
+
+func TestDeletePackage(t *testing.T) {
+	testCases := []struct {
+		desc     string
+		testFunc func(*testing.T)
+	}{
+		{
+			desc: "Not Implemented",
+			testFunc: func(*testing.T) {
+				logger := zap.NewNop()
+				p := &PackagesStateProvider{
+					logger: logger,
+				}
+
+				err := p.DeletePackage("name")
+
+				assert.ErrorContains(t, err, "method not implemented")
+			},
+		},
+	}
+
+	for _, tc := range testCases {
+		t.Run(tc.desc, tc.testFunc)
+	}
+}
+
+func TestLastReportedStatuses(t *testing.T) {
+	pkgName := "observiq-otel-collector"
+	testCases := []struct {
+		desc     string
+		testFunc func(*testing.T)
+	}{
+		{
+			desc: "File doesn't exist",
+			testFunc: func(*testing.T) {
+				noExistYaml := "garbage.yaml"
+				logger := zap.NewNop()
+				p := &PackagesStateProvider{
+					logger:   logger,
+					yamlPath: noExistYaml,
+				}
+
+				actual, err := p.LastReportedStatuses()
+
+				assert.NoError(t, err)
+				assert.Nil(t, actual.ServerProvidedAllPackagesHash)
+				assert.Equal(t, "", actual.ErrorMessage)
+				assert.Equal(t, 1, len(actual.Packages))
+				assert.Equal(t, pkgName, actual.Packages[pkgName].GetName())
+				assert.Equal(t, version.Version(), actual.Packages[pkgName].GetAgentHasVersion())
+				assert.Nil(t, actual.Packages[pkgName].GetAgentHasHash())
+				assert.Equal(t, "", actual.Packages[pkgName].GetServerOfferedVersion())
+				assert.Nil(t, actual.Packages[pkgName].GetServerOfferedHash())
+				assert.Equal(t, protobufs.PackageStatus_Installed, actual.Packages[pkgName].GetStatus())
+				assert.Equal(t, "", actual.Packages[pkgName].GetErrorMessage())
+			},
+		},
+		{
+			desc: "Problem reading file",
+			testFunc: func(*testing.T) {
+				cantReadYaml := "../testdata/noread.yaml"
```

Comment:
So you don't have to remove the written file you can create a temp dir that gets cleaned up at the end of the test by using [t.TempDir](https://pkg.go.dev/testing#T.TempDir).

Then creating the path by using `filepath.Join` so it'll format the path based on the OS running the tests

```go
tmpDir := t.TempDir()
can'tReadYaml := filepath.Join(tmpDir, "noread.yaml")
```

Reply:
@cpheps awesome. good to know